definitions/artifacts/azure-databricks-workspace.json

```diff
@@ -0,0 +1,99 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://schemas.massdriver.cloud/definitions/artifacts/azure-databricks-workspace.json",
+  "$md": {
+    "access": "public"
+  },
+  "type": "object",
+  "title": "Azure Databricks Workspace",
+  "description": "",
+  "additionalProperties": false,
+  "required": [
+    "data",
+    "specs"
+  ],
+  "properties": {
+    "data": {
+      "title": "Artifact Data",
+      "type": "object",
+      "required": [
+        "infrastructure",
+        "security"
+      ],
+      "properties": {
+        "infrastructure": {
+          "title": "Infrastructure configuration",
+          "type": "object",
+          "required": [
+            "ari",
+            "url",
+            "storage_account_name",
+            "raw_container",
+            "scratch_container",
+            "enriched_container"
+          ],
+          "properties": {
+            "ari": {
+              "$ref": "../types/azure-resource-id.json"
+            },
+            "url": {
+              "title": "Azure Databricks workspace URL",
+              "$ref": "../types/https-url.json"
+            },
+            "storage_account_name": {
+              "title": "Storage Account Name",
+              "type": "string",
+              "description": "The name of the storage account where the raw, scratch, and enriched data will be stored in separate containers."
+            },
+            "raw_container": {
+              "title": "Raw Container",
+              "description": "The name of the container where the raw data will be stored.",
+              "$ref": "../types/azure-infrastructure-storage-container.json"
+            },
+            "scratch_container": {
+              "title": "Scratch Container",
+              "description": "The name of the container where intermediate data transformations or experimental analyst data will be stored.",
+              "$ref": "../types/azure-infrastructure-storage-container.json"
+            },
+            "enriched_container": {
+              "title": "Enriched Container",
+              "description": "The name of the container where curated data assets will be stored.",
+              "$ref": "../types/azure-infrastructure-storage-container.json"
+            }
+          }
+        },
+        "security": {
```

Comment:
oooo, I'm liking what I see here but we need to rework it a bit.

```
"security": {
  "$ref": "../types/azure-security.json"
}
```

The structure for the security block is roughly

```
"security": {
  "iam": {
    "any-key-name-but-usually..": {},
    "read/write": {},
    "subscriber": {
      "role": "rolestring",
      "scope": "scopstring"
    }
  }
}
```

If role and scope aren't the right things for Azure now is the time to adjust `em! I believe this is the first bundle that will have the security block for Azure. Is that correct @mclacore ?

Check out the json schema for a GCP artifact that has this block.

https://github.com/massdriver-cloud/artifact-definitions/blob/main/definitions/artifacts/gcp-bucket-https.json#L30-L32

The rest of this looks pretty good but we probably want this to live in the `_artifact.tf`.

```terraform
security = {
  iam = {
    data-write = {
      role = "raw_data_writer_service_principal thing?"
      scope = ""
    }
    data-process = {}
    data-asset-consumer = {}
  }
}
```

@chrisghill this is more motivation for having a description in each iam  block... a lot of this context is lost, thoughts?

Lastly, here's the GCP bundle Terraform.

https://github.com/massdriver-cloud/gcp-gcs-bucket-static-assets/blob/main/src/_artifacts.tf#L7-L20

```terraform
iam = {
  read = {
    role      = "roles/storage.objectViewer"
    condition = "resource.name.startsWith(\"projects/_/buckets/${google_storage_bucket.main.name}\")"
  }
  "read/write" = {
     role      = "roles/storage.objectCreator"
     condition = "resource.name.startsWith(\"projects/_/buckets/${google_storage_bucket.main.name}\")"
  }
  admin = {
    role      = "roles/storage.objectAdmin"
    condition = "resource.name.startsWith(\"projects/_/buckets/${google_storage_bucket.main.name}\")"
  }
}
```

Reply:
> I believe this is the first bundle that will have the security block for Azure. Is that correct @mclacore ?
Correct, I haven't needed the security block yet

Reply:
> @chrisghill this is more motivation for having a description in each iam block... a lot of this context is lost, thoughts?

I like the idea of descriptions, but I think @coryodaniel had a point about the description being in the guide instead of in the artifact. I think this is something we can revisit as we get more users using application creation process in the CLI/UI. But for now I think we should avoid having "metadata" type fields in the artifacts.

Reply:
Yeah we should document them in the guide. That’s what users will see. We should _definitely_ doc then though. Knowledge is power 🧠  💪🏽

Reply:
Ooo I like the sound of that. That would keep the guide in-sync with the def instead of human-mapping.